### PR TITLE
XRM - Updated .then definition to include errorCode and message as parameters in the definition.

### PR DESCRIPTION
**Pull request by carltoncolter**
Updated then callback to include the errorcode and mesage according to the api documentation.

Please fill in this template.

- [X ] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X ] Provide a URL to documentation or source code which provides context for the suggested changes: [Refresh API Documentation](https://msdn.microsoft.com/en-us/library/dn481607.aspx)
- [X] Increase the version number in the header if appropriate. - No
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. - Not substantial

### CODE REVIEW

File: types/xrm/index.d.ts [hunk 1]
```diff
@@ -1256,11 +1256,15 @@ declare namespace Xrm {
             then<U>(onFulfilled?: (value: T) => U | PromiseLike<U>, onRejected?: (error: any) => U | PromiseLike<U>):
                 PromiseLike<U>;
             then<U>(onFulfilled?: (value: T) => U | PromiseLike<U>, onRejected?: (error: any) => void): PromiseLike<U>;
-
+	    // https://msdn.microsoft.com/en-us/library/dn481607.aspx shows errorCode and message
+            then<U>(onFulfilled?: (value: T) => U | PromiseLike<U>, onRejected?: (errorCode: string, message: string) => U | PromiseLike<U>):
```
Comment (daryllabar) [inline types/xrm/index.d.ts]: Error Code is listed as a number in the docs...

File: types/xrm/index.d.ts [hunk 2]
```diff
@@ -1256,11 +1256,15 @@ declare namespace Xrm {
...
+            PromiseLike<U>;
+        
             /**
              * UNDOCUMENTED (Web Client only) Add handlers to be called when the Deferred object is rejected.
              */
             fail<U>(onRejected?: (reason: ErrorResponse) => U | PromiseLike<U>): PromiseLike<U>;
+            fail<U>(onRejected?: (errorCode: string, message: string) => U | PromiseLike<U>): PromiseLike<U>;
```
Comment (daryllabar) [inline types/xrm/index.d.ts]: Here too

**Review by daryllabar — CHANGES_REQUESTED** [review]
Doc lists error code as a number, not a string.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**